src/index.ts

```diff
@@ -1264,17 +1265,17 @@ export class KycDao extends ApiBase {
   @Catch()
   public async switchProviderNetwork(blockchainNetwork: BlockchainNetwork): Promise<void> {
     if (!this._chainAndAddress) {
-      throw new Error('Wallet connection required.');
+      throw new StatusError('WalletNotConnected');
     }
 
     if (!this.blockchainNetworks.includes(blockchainNetwork)) {
-      throw Error('Blockchain network not enabled.');
+      throw new StatusError('NetworkNotEnabled', 'Blockchain network not enabled.');
     }
 
     const networkDetails = this.blockchainNetworkDetails[blockchainNetwork];
 
     if (this._chainAndAddress.blockchain !== networkDetails.blockchain) {
-      throw new Error('Blockhain network is not supported by current wallet provider.');
+      throw new StatusError('NetworkNotSupported');
```

Comment:
Hmm, this probably shouldn't be the same as the SDK not supporting a network. To be honest I'm not even sure what this error is or if my original description was correct...